core/execution/liquidation/engine.go

```diff
@@ -0,0 +1,307 @@
+		return nil, nil
+	}
+	// round up, avoid a value like 0.1 to be floored, favour closing out a position of 1 at least
+	maxCons := uint64(num.DecimalFromFloat(float64(available)).Mul(e.cfg.MaxFractionConsumed).Ceil().IntPart())
+	if maxCons < size {
+		size = maxCons
+	}
+	// get the block hash
+	_, blockHash := vegacontext.TraceIDFromContext(ctx)
+	idgen := idgeneration.New(blockHash + crypto.HashStrToHex("networkLS"+e.mID))
+	// set time for next order, if the position ends up closed out, then that's fine
+	// we'll remove this time when the position is updated
+	if size == 0 {
+		return nil, nil
+	}
+	e.nextStep = now.Add(e.cfg.DisposalTimeStep)
+	// place order using size
+	return &types.Order{
+		ID:          idgen.NextID(),
+		MarketID:    e.mID,
+		Party:       types.NetworkParty,
+		Side:        side,
+		Price:       price,
+		Size:        size,
+		Remaining:   size,
+		TimeInForce: types.OrderTimeInForceIOC,
+		Type:        types.OrderTypeLimit,
+		CreatedAt:   now.UnixNano(),
+		Status:      types.OrderStatusActive,
+		Reference:   "LS", // Liquidity sourcing
+	}, nil
+}
+
+// ClearDistressedParties transfers the open positions to the network, returns the market position events and party ID's
+// for the market to remove the parties from things like positions engine and collateral.
+func (e *Engine) ClearDistressedParties(ctx context.Context, idgen IDGen, closed []events.Margin, mp, mmp *num.Uint) ([]events.MarketPosition, []string, []*types.Trade) {
+	if len(closed) == 0 {
+		return nil, nil, nil
+	}
+	// netork is most likely going to hold an actual position now, let's set up the time step when we attempt to dispose
+	// of (some) of the volume
+	if e.pos.open == 0 || e.nextStep.IsZero() {
+		e.nextStep = e.tSvc.GetTimeNow().Add(e.cfg.DisposalTimeStep)
+	}
+	mps := make([]events.MarketPosition, 0, len(closed))
+	parties := make([]string, 0, len(closed))
+	// order events here
+	orders := make([]events.Event, 0, len(closed)*2)
+	// trade events here
+	trades := make([]events.Event, 0, len(closed))
+	netTrades := make([]*types.Trade, 0, len(closed))
+	now := e.tSvc.GetTimeNow()
+	for _, cp := range closed {
+		e.pos.open += cp.Size()
+		// get the orders and trades so we can send events to update the datanode
+		o1, o2, t := e.getOrdersAndTrade(cp, idgen, now, mp, mmp)
+		orders = append(orders, events.NewOrderEvent(ctx, o1), events.NewOrderEvent(ctx, o2))
+		trades = append(trades, events.NewTradeEvent(ctx, *t))
+		netTrades = append(netTrades, t)
+		// add the confiscated balance to the fee pool that can be taken from the insurance pool to pay fees to
+		// the good parties when the network closes itself out.
+		mps = append(mps, cp)
+		parties = append(parties, cp.Party())
+	}
+	// send order events
+	e.broker.SendBatch(orders)
+	// send trade events
+	e.broker.SendBatch(trades)
+	// the network has no (more) remaining open position -> no need for the e.nextStep to be set
+	if e.pos.open == 0 {
+		e.nextStep = time.Time{}
```

Comment:
Don't know if this goes to the snapshot, but it doesn't marshal well if I remember properly.

Reply:
Haven't encountered any issues TBH, but I believe they're converted to `UnixNano()` for the marshalling